lib/api/1.1/northbound/login.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express'),
+    passport = require('passport');
+
+module.exports = loginRouterFactory;
+
+di.annotate(loginRouterFactory, new di.Provide('Http.Api.Login'));
+di.annotate(loginRouterFactory, new di.Inject(
+        'Auth.Services'
+    )
+);
+
+function loginRouterFactory (
+    authService
+) {
+    var BAD_REQUEST_STATUS = 400;
+    var UNAUTHORIZED_STATUS = 401;
+    var ERROR_STATUS = 500;
+
+    var router = express.Router();
+
+    /**
+     * @api {post} /api/1.1/login/ POST /
+     * @apiVersion 1.1.0
+     * @apiDescription get a token before accessing other apis
+     * @apiName login-post
+     * @apiGroup login
+     * @apiParam {String} username username used to login
+     * @apiParam {String} password password used to login
+     * @apiSuccess {json} a token to be used when accessing other apis:
+     *      {
+     *       "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiY
+     *       WRtaW4iLCJpYXQiOjE0NTI0ODA3MTIsImV4cCI6MTQ1MjU2NzExMn0.Qo-9y
+     *       omlkd0WE1jlEA-L1WqA6k_9qJYhkicQL5IUmSg"
+     *      }
+     */
+
+    router.post('/login',
+        function(req, res, next) {          //Middleware to do authentication.
```

Comment:
Since this anonymous callback function is a little complex, but give the callback a name will help debugging, refer to this link: [You Don't Know JS: Scope & Closures - Anonymous vs. Named] (https://github.com/getify/You-Dont-Know-JS/blob/master/scope%20&%20closures/ch3.md#anonymous-vs-named)